File: Iteration2/code/CSE3063GRP12/src/core/general_providers/Logger.java
package core.general_providers;

import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.Files;
import java.nio.file.StandardOpenOption;
import java.io.IOException;



public class Logger {

    private Path filePath;
    private String className;
    private String methodName;
    public Logger(String path){
        filePath = Paths.get(path);
        className = "";
        methodName = "";
        try {
            // Create the file if it doesn't exist
            if (!Files.exists(filePath)) {
                Files.createFile(filePath);
            }
        }catch (IOException e){}
    }
    public Logger(){
        filePath = Paths.get("./Iteration2/code/CSE3063GRP12/database/log/log.txt");
        try {
            // Create the file if it doesn't exist
            if (!Files.exists(filePath)) {
                Files.createFile(filePath);
            }
        }catch (IOException e){}
    }
    public void write(String input) {
        try {
            StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();

            if (stackTrace.length >= 5) {
                StackTraceElement caller = stackTrace[4];
                className = caller.getClassName();
                methodName = caller.getMethodName();}

            // Write content to the file
            String message = input + " is user's choice in method " + methodName + " in class " + className + System.lineSeparator();
            Files.write(filePath, message.getBytes(), StandardOpenOption.APPEND);

        } catch (IOException e) {}
    }
    public void writeException(String input) {
        try {
            Files.write(filePath, (input + System.lineSeparator()).getBytes(), StandardOpenOption.APPEND);
        }catch (IOException e){}
    }
}

File: Iteration3/code/CSE3063GRP12/src/core/general_providers/Logger.py
import os
import traceback
from AppConstant import AppConstant

class Logger:

    def __init__(self, path=None):
        if path is None:
            path = AppConstant.getInstance().get_base_path()+"log"+"log.txt"
        self.__filePath = path
        self.__className = ""
        self.__methodName = ""
        try:
            if not os.path.exists(self.__filePath):
                open(self.__filePath, 'w').close()
        except IOError:
            pass

    def write(self, input_text):
        try:
            stack_trace = traceback.extract_stack()

            if len(stack_trace) >= 3:
                caller = stack_trace[-3]
                self.__className = caller.filename
                self.__methodName = caller.name

            # Write content to the file
            message = f"{input_text} is user's choice in method {self.__methodName} in class {self.__className}\n"
            with open(self.__filePath, 'a') as file:
                file.write(message)

        except IOError:
            pass

    def write_exception(self, input_text):
        try:
            with open(self.__filePath, 'a') as file:
                file.write(f"{input_text}\n")
        except IOError:
            pass
